Integrations/integration-IPInfo_PY.yml

```diff
@@ -0,0 +1,237 @@
+    BASE_URL = 'http://ipinfo.io/'
+    USE_SSL = not demisto.params().get('insecure', False)
+    APIKEY = demisto.params()['apikey']
+    IP = demisto.args().get('ip')
+    FIELD = demisto.args().get('field')
+
+    headers = {
+        'Content-Type': "application/json",
+        'Accept': "application/json"
+        }
+
+    ## Helper Functions
+
+    ## Sends Request to IP Info
+    def send_request():
+        if APIKEY is None:
+            r = requests.request("GET",BASE_URL+IP+'/json',headers=headers)
+        else:
+            r = requests.request("GET",BASE_URL+IP+'/json?token='+APIKEY,headers=headers)
+        data = r.json()
+        return data
+
+    # Checks to see if Request returned an error message
+    def error_check(data):
+        if 'error' in data:
+            errormsg = data['error']['message']
+            md = '## There has been an error: '+errormsg+''
+            # Alerts user if there was an error
+            demisto.results({
+                'Type': entryTypes['note'],  # note, warning, error
+                'ContentsFormat': formats['markdown'],  # json, markdown, file, text
+                'Contents': md,
+                'HumanReadable':md
+            })
+            return True
+        else:
+            return False
+
+
+    #  FUNCTIONS
+
+    def ipinfo_command():
+        data = send_request()
+        errors = error_check(data)
+
+        # Check for errors
+        if errors is False:
+            loc_unformatted = data.get('loc', '1,2')
+
+            # Let's build a JSON for a MAP!
+            _lat,_long = loc_unformatted.split(',')
+            map_dict = {'lat': _lat,'lng': _long}
+            map_json = json.dumps(map_dict)
+
+            default_value = 'Not found'
+
+            # Mapping Returned Values from Response
+            table = [{'Hostname': data.get('hostname', default_value),
+                'City': data.get('city', default_value),
+                'Region': data.get('region', default_value),
+                'Country': data.get('country', default_value),
+                'Postal': data.get('postal', default_value),
+                'Phone': data.get('phone', default_value),
+                'Organization': data.get('org', default_value),
+                'Latitude': _lat,
+                'Longitude': _long
+            }]
+            # Building Context JSON
+            ec = {
+                'IP':
+                    { 'Address': IP,
+                      'Hostname': data.get('hostname', default_value),
+                      'ASN': data.get('org', default_value),
+                      'Geo': {
+                          'Location': data.get('loc', default_value),
+                          'Country': data.get('country', default_value),
+                          'Description': '"'+data.get('city', default_value) + ', ' + data.get('region', default_value) + ', ' + data.get('postal', default_value) + ', ' + data.get('country', default_value)+'"'}},
```

Comment:
this could result in:
`"Not found, Not found, Not found, Not found"`
try thinking of a better way (or consult with me)